test/evaluate_integration_test.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2024 Searis AS
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	clarify "github.com/clarify/clarify-go"
+	"github.com/clarify/clarify-go/fields"
+)
+
+func TestEvaluate(t *testing.T) {
+	ctx := context.Background()
+	creds := getDefaultCreds()
+	client := creds.Client(ctx)
+	prefix := createPrefix()
+	// ---
+	a := Args{
+		ctx:    ctx,
+		creds:  creds,
+		client: client,
+		prefix: prefix,
+	}
+
+	svelg(a, insertDefault)
+	svelg(a, saveSignalDefault)
+	svelg(a, publishSignalDefault)
+
+	got, err := evaluateDefault(a)
+
+	if err != nil {
+		t.Errorf("%v", err)
+	} else {
+		jsonEncode(t, got)
+	}
+}
+
+func evaluateDefault(a Args) (*clarify.EvaluateResult, error) {
+	selectItemResponse, err := selectItemDefault(a)
+	if err != nil {
+		panic(err)
+	}
+
+	itemIDs := make([]string, 0, len(selectItemResponse.Data))
+	for _, d := range selectItemResponse.Data {
+		itemIDs = append(itemIDs, d.ID)
+	}
+
+	query := createQuery(a.prefix)
+
+	return evaluate(a.ctx, a.client, itemIDs, query)
+}
+
+func evaluate(ctx context.Context, client *clarify.Client, itemIDs []string, query fields.ResourceQuery) (*clarify.EvaluateResult, error) {
```

Comment:
Can we actually do _without_ the `evaluate` helper method? It contains lots of hard-coded values and doesn't have a _clear_ return value. Instead, you should look at declaring tests using a private testCase type and test function:

```go

func TestFoo(t *testing.T) {
    var (
       // define variables/constants for all test.
    )
    type testCase struct{
       // define test case parameters for the test.
    }
    
    // setup things common to all tests, e.g. client connection (must not be mutated by tests!).
    
    test := func(tc testCase) func(t *testing.T) {
         return func(t *testing.T) {
             t.Helper() // Errors are reported at the line where the function is called, and not at the failing line inside the function.
             // set-up things specific for the test, often using helper methods.
             
             // run test using parameters from tc.
         }
    }
    t.Run("Describe test case", test(testCase{...})
    t.Run("Describe other test case", test(testCase{...})
}
```

With a `TestEvaluate`, then you would ensure that all setup and input parameters for the test are provided  in the `testCase` type, and simply do the evaluate call inside the `test` function.